app/models/user.rb

```diff
@@ -4,6 +4,20 @@ class User < ApplicationRecord
 
   class UserAuthenticationException < StandardError; end
 
+  EMAIL_DOMAIN_DENYLIST = [
+    "dwp.gov.uk",
```

Comment:
It would be nice to be able to test this locally/in dev, but without having to use an email address with a domain in this list.

If we added `digital.cabinet-office.gov.uk` to this list when `Rails.env.development? || Settings.forms_env.to_sym == :development`, we could verify the behaviour by trying to sign in with a `+` email address.

Reply:
Hmm interesting suggestion. Wouldn't this make it harder for new team members to use dev, as their actual email address would be blocked by default? Or am I missing something that means our normal email addresses wouldn't be blocked?

Reply:
@jamie-o-wilkinson yes it would make it harder for new team members to use dev... I couldn't think of another way of doing it though 😅 The developer strategy sounds nice though, I wonder if it's worth pulling that out into a separate PR @thomasiles?

Reply:
I'm going to merge this PR and then look at this afterwards!